src/classes/frModel.cls

```diff
@@ -0,0 +1,71 @@
+public abstract class frModel {
+	protected abstract SObject getObject();
+    public abstract List<frMapping__c> getMappings();
+
+    public virtual void populateFromRequest(Map<String, Object> request) {
+    	Map<String, Schema.SObjectField> fields = frSchemaUtil.getFields(getObject().getSObjectType().getDescribe().getName());
+
+        Map<String, frMapping__c> frNameToSfName = new Map<String, frMapping__c>();
+        for(frMapping__c mapping : getMappings()) {
+            frNameToSfName.put(mapping.fr_Name__c, mapping);
```

Comment:
And yeah I think as far as looking at the different fields, I'm not sure we can do much about it in code (like if we try to query for the old version of the field but that field doesn't even exist I think we'd get some SF exceptions), so yeah I think it is a you gotta data loader the data